Assets/Scripts/Battlefield/CreatureScripts/MovementSystem.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+
+namespace SwordAndBored.Battlefield.CreaturScripts
+{
+    public class MovementSystem : MonoBehaviour
```

Comment:
I was thinking of having something like 
``` 
IMovementSystem movementSystem = new AStarMovementSystem();
```
in Unit or Creature and running
```
activePlayer.Move(destinationCoordinates);
```
in your movement state which would be a method that looks like
```
movementSystem.FindPathAndMove(agent, currentCoordinates, destinationCoords);
```
and FindPathAndMove would have the LineRenderer and AStar Module inside it that it controls.

This allows the user to never have to see the LineRenderer details and AStar details in the stateMachine but just a Move() method on the ActivePlayer which internally deals with movement according to the MovementSystem that player is instantiated with. Similar to how the Brain gets passed around inside the state machine but using the get component and it has all the data and methods it requires and the different states just call it and dont do much else but instead call the methods and use the data inside the Brain.

Reply:
This also keeps movementSystem away from MonoBehavior which is unnecessary since the StateMachine and Unit/Creature tell it when to do what and an Update and Start and Awake is just unnecessary.